### DISCUSSION TITLE
[Kavita] New library type for non-series files

### DISCUSSION THREAD

**Discussion by DieselTech** [Kareadita/Kavita#2544] [Ideas]
### Idea Description

Posted by Argonel
· Oct 2022
· 23 Upvotes
https://feats.kavitareader.com/posts/222/new-library-type-for-non-series-files

Description

Can we a have a library type that collects the files from a directory and presents them as a group, but does not try to parse them into a series. I would use this for books that have a related topic, but have no particular reading order.

Presentation could be similar to specials, but when a book is completed the reader should return to the book display, not move on to the "next" book

### Duration of Using Kavita

42

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by LnLcFlx** [comment]
Sadly, this is the only reason why I am not yet able to use Kavita.
I think there are too many cases where a non-default directory structure makes sense that I don't think one should force a specific structure on the user.

- I also group some books by topic like @DieselTech (namely science textbooks).
- In addition, I keep all fiction e-books in a single flat folder because this is easier to maintain for me on my e-reader that I have it synced with. I rarely have authors with multiple books thus grouping them by author would become unnecessarily cluttered.
- What's more, I'd also like to use Kavita to read newspapers and magazines I have downloaded. The corresponding folder structure that makes most sense to me has lots of layers because one newspaper brand might have multiple newspapers with special issues and whatnot. I'd like to have control over how they get sorted into series (only the top-level folders or only the bottom-level folders for example).

For inspiration one could take a look at how [stump](https://github.com/stumpapp/stump) (which is the only option that works for me right now) [solved this](https://www.stumpapp.dev/guides/basics/libraries).
I get if this is hard to implement and to maintain, but it would be nice to have at least some more flexibility even if it is only targeted for experienced users.

**Comment by rix8675** [comment]
i've also had to workaround this with wierd ways even developed a script to edit comicinfo to add the info i want and use calibre to mass edit epubs.
